### ISSUE TITLE
Scalatest mixin to override the behavior of `Matchers`?

### ISSUE THREAD

**Issue by Daenyth** [softwaremill/diffx#49]
In our codebase we use the scalatest `Matchers` trait pretty heavily.

Is it possible to extend that trait and override the code in it such that the interface remains the same but the errors are reported using diffx? I could add an extra mixin to a base test class, but changing every single test file in the project is awkward. I don't want to "just" add the `DiffMatcher` trait ad-hoc because the codebase already suffers from "too many ways to assert" problems in consistency.

**Comment by ghostbuster91** [comment]
So you basically want to redirect `shouldBe` method to diffx, is that right?

**Comment by Daenyth** [comment]
yes, `shouldBe`, `shouldEqual`, `should be()`, and so on. It would be a huge bonus to also get some of the collection ones like "same elements" or "same elements in order", but that's a lot less important for me

**Comment by Daenyth** [comment]
If this is plausible to implement I'm happy to POC it in my codebase and then extract it for a PR to you, I just wasn't sure if things were structured in a way that makes it possible to do

**Comment by ghostbuster91** [comment]
Sorry for the delay, but recently I am short on time. I am not sure if it is possible, I have one idea which I would like to test today and I will let you know the results.

**Comment by ghostbuster91** [comment]
Ok, so `shouldBe` and `shouldEqual` are easy, you just have to write an implicit class as follows:
```
  implicit class ShouldBeWrapper[T](t: T) {
    def shouldBe(t2: T)(implicit d: Diff[T]): Assertion = t should matchTo(t2)
  }
```
`should + matcher` is a little bit more tricky. You can shadow the `be` word by specifying a `be` function: 
```
  def be[T: Diff](t: T) = DiffForMatcher(t)
```
The rest should be analogical. Let me know if that helped
